File: PegawaiApp/app/src/main/java/com/udacoding/pegawaiapp/splash/SplashActivity.java
package com.udacoding.pegawaiapp.splash;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

import com.udacoding.pegawaiapp.R;
import com.udacoding.pegawaiapp.home.MainActivity;
import com.udacoding.pegawaiapp.login.LoginActivity;
import com.udacoding.pegawaiapp.utils.SessionManager;

public class SplashActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        final SessionManager sessionManager = new SessionManager(this);
        Handler handler = new Handler();

        handler.postDelayed(new Runnable() {
            @Override
            public void run() {

                if(sessionManager.isLogin()){

                    startActivity(new Intent(SplashActivity.this, MainActivity.class));
                }
                else{
                    startActivity(new Intent(SplashActivity.this, LoginActivity.class));

                }

                finish();


            }
        },4000);
    }
}

File: udacodingecommerce/app/src/main/java/com/udacoding/udacodingecommerce/ui/splash/SplashActivity.kt
package com.udacoding.udacodingecommerce.ui.splash

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.Handler
import com.udacoding.udacodingecommerce.ui.main.MainActivity
import com.udacoding.udacodingecommerce.R
import com.udacoding.udacodingecommerce.ui.login.LoginActivity
import com.udacoding.udacodingecommerce.util.SessionManager
import org.jetbrains.anko.startActivity

class SplashActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)

        var sesi = SessionManager(this)


        Handler().postDelayed(Runnable {

            if(sesi.isLogin) startActivity<MainActivity>()
            else startActivity<LoginActivity>()
            finish()

        },4000)


    }
}
